Add 'Updating the boot loader on RHCOS nodes using bootupd' to update section & fix issues with the bootupd page (#71174)
openshift/openshift-docs: modules/architecture-rhcos-updating-bootloader.adoc

[id="architecture-rhcos-updating-bootloader.adoc_{context}"]
= Updating the bootloader using bootupd

> To update the bootloader by using `bootupd`, you must either run the `bootupctl update` command on {op-system} machines manually or provide a machine config with a `systemd` unit.

Reviewer: ```suggestion
To update the boot loader by using `bootupd`, you must either run the `bootupctl update` command on {op-system} machines manually or provide a machine config with a `systemd` unit.
```

Unlike `grubby` or other bootloader tools, `bootupd` does not manage kernel space configuration such as passing kernel arguments.

> To configure kernel arguments, see xref:../../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-kernel-arguments_nodes-nodes-managing[Adding kernel arguments to Nodes].

Reviewer: ```suggestion
To configure kernel arguments, see xref:../../nodes/nodes/nodes-nodes-managing.adoc#nodes-nodes-kernel-arguments_nodes-nodes-managing[Adding kernel arguments to nodes].
```

[NOTE]
====

> You can use `bootupd` to update the bootloader to protect against the BootHole vulnerability.

Reviewer: ```suggestion
You can use `bootupd` to update the boot loader to protect against the BootHole vulnerability.
```

====

.Manual update method

> You can manually inspect the status of the system and update the bootloader by using the `bootupctl` command-line tool.

Reviewer: ```suggestion
You can manually inspect the status of the system and update the boot loader by using the `bootupctl` command-line tool.
```

----

.Machine config method

> Another way to automatically update the bootloader during the boot process using `bootupd` is by providing a machine config.

Reviewer: ```suggestion
Another way to automatically update the boot loader during the boot process using `bootupd` is by providing a machine config.
```

* Provide a machine config file with the `systemd` unit, as shown in the following example:
+
.Example output

> [source,yaml]

Reviewer: the attribute in the code block isn't rendering, you need the substitution:
```suggestion
[source,yaml,subs="attributes+"]
```

----
  variant: rhcos

>   version: v{MaistraVersion}

Author: This has been replaced by `v2.4` and should be `v4.15.0` instead so this is probably not the right attribute.

Reviewer: if this is the RHCOS version then `{product-version}` ?

Author: It's the Butane spec version. I'll have to dig into other Butane example to see what we use.

Author: Hum, my bad, it's not a Butane config but directly a MachineConfig one. Will have to look again.

          [Install]
          WantedBy=multi-user.target
----

[NOTE]
====

> This `systemd` unit will try to update the bootloader on {op-system} machines every boot. Interrupting this process might lead to unbootable nodes.

Reviewer: ```suggestion
This `systemd` unit will try to update the boot loader on {op-system} machines every boot. Interrupting this process might lead to unbootable nodes.
```
